rotation.py

```diff
@@ -238,25 +247,36 @@ def print_state(val):
     print_state(os.environ.get("PERF_TRIAGE_BOT_CACHED_USER_SECRETS"))
 
 
+def get_week(date):
+    week = datetime.fromisocalendar(*date.isocalendar()[:2], day=1)
+    return week.strftime("%Y-%m-%d")
+
+
 def main():
     args = parse_args()
 
     rotations = load_rotations()
     leaders = [m for m in MEMBERS if m.lead]
     while len(rotations) < len(leaders):
         # create some history to improve selection
-        rotations.append(generate_rotation(leaders, rotations))
+        week = get_week(DATE - timedelta(weeks=(len(leaders) - len(rotations))))
+        rotations.setdefault(week, generate_rotation(leaders, rotations))
 
-    # generate new rotation
-    rotations.append(generate_rotation(leaders, rotations))
+    print(f"Generated on {DATE}")
 
-    print(f"Generated on {DATE}\n")
+    print("\nThis week:")
+    this_week = get_week(DATE)
+    rotations.setdefault(this_week, generate_rotation(leaders, rotations))
+    print(f"{this_week}: {rotations[this_week]}")
 
-    print(f"This week: {rotations[-2]}")
-    print(f"Next week: {rotations[-1]}")
+    print("\nNext week:")
+    next_week = get_week(DATE + timedelta(weeks=1))
+    rotations.setdefault(next_week, generate_rotation(leaders, rotations))
+    print(f"{next_week}: {rotations[next_week]}")
 
-    print("\nPrevious rotations:")
-    [print(r) for r in reversed(rotations[:-2])]
+    print("\nHistory:")
+    for week, rotation in list(rotations.items())[2:]:
```

Comment:
Is there any guarantee that `rotations.items()` is sorted? I think we want to wrap this in `sorted(...)`